### DISCUSSION TITLE
Mapping Error after fresh 2.3.40 install

### DISCUSSION THREAD

**Discussion by Cyb3rtus** [Security-Onion-Solutions/securityonion#3816] [Unsupported Versions]
Hi,

I have installed a new instance (three times already too reproduce the errors) of SO 2.3.40 using the ISO. Everything went fine during installation. However I immediatly noticed several errors in Kibana. First of all some dashboards have errors like: "Could not locate that index-pattern-field (id: notice.message.keyword)" in dashboard Security Onion - Zeek - Notices. 

Also when viewing the index mapping it tells me there are mapping errors: 
"Mapping conflict
5 fields are defined as several types (string, integer, etc) across the indices that match this pattern. You may still be able to use these conflict fields in parts of Kibana, but they will be unavailable for functions that require Kibana to know their type. Correcting this issue will require reindexing your data."

Now I know there are fixes and that you can re-index, but this does not seem like normal behaviour and I would like to know if other people have experienced these issues and might know a fix or how to prevent this issue. 

{
  "took": 43,
  "timed_out": false,
  "_shards": {
    "total": 7,
    "successful": 6,
    "skipped": 0,
    "failed": 1,
    "failures": [
      {
        "shard": 0,
        "index": "securityonion:so-ossec",
        "node": "_2y3WhWsTTO3uVfFn2e8og",
        "reason": {
          "type": "illegal_argument_exception",
          "reason": "Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [rule.uuid] in order to load field data by uninverting the inverted index. Note that this can use significant memory."
        }
      }
    ]
  },
  "_clusters": {
    "total": 1,
    "successful": 1,
    "skipped": 0
  },
  "hits": {
    "total": 473,
    "max_score": null,
    "hits": []
  },
  "aggregations": {
    "2": {
      "doc_count_error_upper_bound": 0,
      "sum_other_doc_count": 0,
      "buckets": [
        {
          "key": "5501",
          "doc_count": 122
        },
        {
          "key": "5502",
          "doc_count": 117
        },
        {
          "key": "553",
          "doc_count": 45
        },
        {
          "key": "516",
          "doc_count": 42
        },
        {
          "key": "19009",
          "doc_count": 40
        },
        {
          "key": "19008",
          "doc_count": 21
        },
        {
          "key": "550",
          "doc_count": 15
        },
        {
          "key": "533",
          "doc_count": 14
        },
        {
          "key": "5402",
          "doc_count": 13
        },
        {
          "key": "19007",
          "doc_count": 10
        },
        {
          "key": "502",
          "doc_count": 7
        },
        {
          "key": "503",
          "doc_count": 7
        },
        {
          "key": "5715",
          "doc_count": 7
        },
        {
          "key": "5503",
          "doc_count": 3
        },
        {
          "key": "2501",
          "doc_count": 2
        },
        {
          "key": "501",
          "doc_count": 2
        },
        {
          "key": "5557",
          "doc_count": 2
        },
        {
          "key": "510",
          "doc_count": 1
        },
        {
          "key": "5403",
          "doc_count": 1
        },
        {
          "key": "554",
          "doc_count": 1
        },
        {
          "key": "5902",
          "doc_count": 1
        }
      ]
    }
  }
}

**Comment by sanctanox** [comment]
Sounds fairly similar to what I'm experiencing as well.  Are you using the Wazuh agent on a Windows server?

https://github.com/Security-Onion-Solutions/securityonion/discussions/3825

Hoping someone has some ideas that can help.

**Reply by Cyb3rtus** [reply]
I am actually, just the one agent installed.

**Comment by defensivedepth** [comment]
@Cyb3rtus Can you please show the output of the following?

`curl -k  https://localhost:9200/_cat/indices | grep ossec`

**Reply by Cyb3rtus** [reply]
@defensivedepth 

Of course, here it is:

green  open so-ossec-2021.04.09             utbvnqBiQCOpweBfxd8uQQ 1 0 39377     0  80.6mb  80.6mb
green  open so-ossec-2021.04.08             4MvRiMuuQHabrhDlIW2MBA 1 0 22485    72  48.2mb  48.2mb
green  open so-ossec-2021.04.13             Gag2tz4MTrm8EA1EMuLY-g 1 0 13119     0  31.9mb  31.9mb
yellow open so-ossec                                 oviKD1h1Sr-qEZL7CFvRvw 1 1 16710     4  33.3mb  33.3mb
green  open so-ossec-2021.04.10             4KF2AIeWRBak2Yh9ID3tiw 1 0 39250     0  80.6mb  80.6mb
green  open so-ossec-2021.04.12             HFYY0Vo-TjCBxhJqTsPsJQ 1 0 39323     0  82.6mb  82.6mb
green  open so-ossec-2021.04.11             4iIsFbT8RF6aFl485sw9Dw 1 0 39318     0  82.6mb  82.6mb

Kibana now says "1 out of 15 shards failed" when opening suricate alerts from the home dashboard in Kibana and there are quite a lot of dashboards where some widgets say: Could not locate that index-pattern-field (id: _some.field_)

Also over the weekend I have installed another fresh SO WITHOUT any agent. That one only has many "Could not locate that index-pattern-field" errors, but so far no errors with shard. The only difference is that I installed a wazuh agent on the one from last week. I used the wazuh agent from the downloads page in SO.

**Reply by defensivedepth** [reply]
@Cyb3rtus Please follow the instructions here - it looks like you are seeing the same bug.

https://github.com/Security-Onion-Solutions/securityonion/discussions/3825#discussioncomment-600584

**Reply by Cyb3rtus** [reply]
Great! This seems to fix the update!
